### PR TITLE
Fix the processing of JSON-RPC requests encoded using the GET method

### PR DESCRIPTION
**Pull request by aplgithub**
When an invocation onto a JSON-RPC endpoint arrives without a ```param``` query parameter on a GET request, the assembly of the JSON-RPC envelope throws an exception that is not easy to understand.  There seem to be other problems with the use of the HTTP GET method as well; these should be resolved in this patch as well.

### CODE REVIEW

File: src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java [hunk 1]
```diff
@@ -159,10 +161,76 @@ private static void loadAnnotationSupportEngine() {
...
+		envelope.append(JSONRPC);
+		envelope.append("\":\"");
+		envelope.append(VERSION);
+		envelope.append("\",\"");
+		envelope.append(ID);
+		envelope.append("\":");
+
+		// the 'id' value is assumed to be numerical.
+
+		if (null!=id && 0!=id.length()) {
```
Comment (gaborbernat) [inline src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java]: add spaces around the !=

File: src/test/java/com/googlecode/jsonrpc4j/server/JsonRpcBasicServerTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package com.googlecode.jsonrpc4j.server;
```
Comment (gaborbernat) [inline src/test/java/com/googlecode/jsonrpc4j/server/JsonRpcBasicServerTest.java]: seems you removed lot of tests in the previous class :) don't like that part, why?

File: src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java [hunk 3]
```diff
@@ -30,14 +30,14 @@
 import java.lang.reflect.Type;
```
Comment (gaborbernat) [inline src/main/java/com/googlecode/jsonrpc4j/JsonRpcBasicServer.java]: please provide a concrete example this fixes :)

**Comment by aplgithub** [comment]
Sorry - ended up in a mess on my branch so I opened another PR.

**Pull request closed without merging**